types/lodash/lodash-tests.ts

```diff
@@ -6068,20 +6156,45 @@ namespace TestMap {
     let array: number[] | null | undefined = [] as any;
     let list: _.List<number> | null | undefined = [] as any;
     let obj: any = {};
-    let dictionary: _.Dictionary<number> | null | undefined = obj;
+    let dictionary: _.Dictionary<number> | null | undefined = anything;
+    let numericDictionary: _.NumericDictionary<number> | null | undefined = anything;
 
     let listIterator: (value: number, index: number, collection: _.List<number>) => AbcObject = (value: number, index: number, collection: _.List<number>) => ({ a: 1, b: "", c: true });
     let dictionaryIterator: (value: number, key: string, collection: _.Dictionary<number>) => AbcObject = (value: number, key: string, collection: _.Dictionary<number>) => ({ a: 1, b: "", c: true });
+    let numericDictionaryIterator: (value: number, key: string, collection: _.NumericDictionary<number>) => AbcObject = (value: number, key: string, collection: _.NumericDictionary<number>) => ({ a: 1, b: "", c: true });
```

Comment:
This style was written by me when I was young and foolish. A simpler way would be:

```ts
const numericDictionaryIterator = (value: number, key: string, collection: _.NumericDictionary<number>): AbcObject => ({ a: 1, b: "", c: true });
```

Or better yet, don't declare a `numericDictionaryIterator` at all, and instead write the function inline (It looks like you've already done that below):

```ts
// $ExpectType ???
_.map(numericDictionary, (value, index, collection) => {
    value; // $ExpectType number
    index; // $ExpectType string
    collection; // $ExpectType NumericDictionary<number>
});
```